pkg/alertmanager/operator.go

```diff
@@ -1790,3 +1796,25 @@ func ListOptions(name string) metav1.ListOptions {
 func tlsAssetsSecretName(name string) string {
 	return fmt.Sprintf("%s-tls-assets", prefixedName(name))
 }
+
+func GetAppliedConfigFromCondition(c *monitoringv1.Condition) *monitoringv1ac.ConditionApplyConfiguration {
+	return monitoringv1ac.Condition().
+		WithType(c.Type).
+		WithStatus(c.Status).
+		WithLastTransitionTime(c.LastTransitionTime).
+		WithReason(c.Reason).
+		WithMessage(c.Message).
+		WithObservedGeneration(c.ObservedGeneration)
+}
+
+func GetAppliedConfigStatusFromStatus(a *monitoringv1.Alertmanager, asac *monitoringv1ac.AlertmanagerStatusApplyConfiguration, conditions ...monitoringv1.Condition) {
+	asac.WithPaused(a.Status.Paused)
+	asac.WithReplicas(a.Status.Replicas)
+	asac.WithAvailableReplicas(a.Status.AvailableReplicas)
+	asac.WithUpdatedReplicas(a.Status.UpdatedReplicas)
+	asac.WithUnavailableReplicas(a.Status.UnavailableReplicas)
+
+	for _, condition := range conditions {
+		asac.WithConditions(GetAppliedConfigFromCondition(&condition))
```

Comment:
(nit) this is what I meant by getting rid of `GetAppliedConfigFromCondition`.
```suggestion
		asac.WithConditions(
			monitoringv1ac.Condition().
			WithType(c.Type).
			WithStatus(c.Status).
			WithLastTransitionTime(c.LastTransitionTime).
			WithReason(c.Reason).
			WithMessage(c.Message).
			WithObservedGeneration(c.ObservedGeneration),
		)		
```